lib/custom/src/MShop/Service/Provider/Payment/Stripe.php

```diff
@@ -32,21 +126,309 @@ protected function getConfigPrefix()
 	}
 
 
+	/**
+	 * Tries to get an authorization or captures the money immediately for the given order if capturing the money
+	 * separately isn't supported or not configured by the shop owner.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order Order invoice object
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Standard Form object with URL, action and parameters to redirect to
+	 *    (e.g. to an external server of the payment provider or to a local success page)
+	 */
+	public function process(\Aimeos\MShop\Order\Item\Iface $order, array $params = [])
+	{
+		if( $this->getValue( 'onsite' ) == true && ( !isset( $params['paymenttoken'] )  ) ) {
+			return $this->getPaymentForm( $order, $params );
+		}
+		return $this->processOrder($order, $params);
+	}
+
+
+	/**
+	 * Returns the payment form for entering payment details at the shop site.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order Order object
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Iface Form helper object
+	 */
+	protected function getPaymentForm(\Aimeos\MShop\Order\Item\Iface $order, array $params)
+	{
+		$list = [];
+		$feConfig = $this->feConfig;
+
+		foreach ($feConfig as $key => $config) {
+			$list[$key] = new \Aimeos\MW\Criteria\Attribute\Standard($config);
+		}
+		$url = $this->getConfigValue(array('payment.url-self'));
+		return new \Aimeos\MShop\Common\Item\Helper\Form\Standard($url, 'POST', $list, false, $this->getHTMLform());
+	}
+
+
 	/**
 	 * Returns the value for the given configuration key
 	 *
 	 * @param string $key Configuration key name
 	 * @param mixed $default Default value if no configuration is found
 	 * @return mixed Configuration value
 	 */
-	protected function getValue( $key, $default = null )
+	protected function getValue($key, $default = null)
+	{
+		switch ($key) {
+			case 'type':
+				return 'Stripe';
+			case 'onsite':
+				return true;
+			case 'token':
```

Comment:
What is "token" for? Isn't it the specific token for the payment of the customer?